src/sparsify/auto/utils/error_handler.py

```diff
@@ -0,0 +1,186 @@
+            exception,
+            torch.distributed.elastic.multiprocessing.errors.ChildFailedError,
+        ):
+            # Grabbing exception only from first worker
+            _, first_error = exception.get_first_failure()
+            if isinstance(first_error.message, dict):
+                message = first_error.message["message"].split(": ")
+                if message[0] in self._python_exceptions:
+                    exception = self._python_exceptions[message[0]](message[1])
+
+        # Check if an out of memory error was thrown by comparing error message with
+        # substrings that are known to represent out of memory errors
+        if isinstance(exception, RuntimeError) and (
+            any(
+                [
+                    memory_substring in exception.args[0]
+                    for memory_substring in MEMORY_ERROR_SUBSTRINGS
+                ]
+            )
+        ):
+            self._caught_memory_errors.append(exception)
+            self._caught_memory_error_on_last_attempt = True
+
+        else:
+            self._caught_runtime_errors.append(exception)
+            self._caught_memory_error_on_last_attempt = False
+
+    def is_memory_error(self) -> bool:
+        """
+        Returns true if the last run raised an out of memory error
+        """
+        return self._caught_memory_error_on_last_attempt
+
+    def raise_exception_summary(self):
+        """
+        Raise an exception that summarizes the exception history for run attempts
+        of this stage only
+        """
+
+        # Run failed due to non-memory related error
+        if len(self._caught_runtime_errors) >= self._max_retry_attempts:
+            first_error = self._caught_runtime_errors[0]
+            # If identical errors raised, print error just once. Otherwise, enumerate
+            # errors in printout.
+            if all(
+                [
+                    (
+                        (type(error) == type(first_error))
+                        and (error.args == first_error.args)
+                    )
+                    for error in self._caught_runtime_errors
+                ]
+            ):
+                raise RuntimeError(
+                    f"Run failed after {len(self._caught_runtime_errors)} attempts "
+                    f"with error: {first_error}"
+                )
+
+            else:
+                error_list = "\n".join(
+                    [
+                        f"Attempt {attempt_num}: {error}"
+                        for attempt_num, error in enumerate(self._caught_runtime_errors)
+                    ]
+                )
+                raise RuntimeError(
+                    f"Run failed after {len(self._caught_runtime_errors)} attempts "
+                    f"with the following errors:\n{error_list}"
+                )
+
+        # Run failed due to memory error that couldn't be overcome
+        elif self._max_memory_stepdowns >= self._max_memory_stepdowns:
```

Comment:
Is this the intention?
```suggestion
        elif len(self._caught_memory_errors) >= self._max_memory_stepdowns:
```

Reply:
Yep, thanks! The result of too many quick changes to one function